### PR TITLE
Get response for date question

### PR DESCRIPTION
**Pull request by leenagupte**
Trello: https://trello.com/c/h5u7N1T9
Follows on from: PR #5315 

# What's changed?

Add a method to `DateQuestionPresenter` to get the current value of the date fields.

## Changes in this PR
- [x] Date question

## Changes in other PRs
- [x] When a user click back on a URL based Smart Answer then the previous questions show, with the previous answer pre-filled (#5315)
- [x] When a user click back on a session based Smart Answer then the previous questions show, with the previous answer pre-filled (#5315)
- [x] Country-select question (#5315)
- [x] Money question (#5315)
- [x] Postcode question (#5315)
- [x] Value question (#5315)
- [x] Checkbox question (#5316)
- [x] Radio button question
- [x] Salary question (#5320)

# Why?

There is currently a bug in session-based smart-answers, where answers to previous date questions are not being populated even though the answers exist in the session. 

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: app/presenters/date_question_presenter.rb [hunk 1]
```diff
@@ -15,15 +13,24 @@ def response_label(value)
   end
 
   def selected_day
-    prefill_value_for(self, :day)
+    return if response_for_current_question.blank?
+    return response_for_current_question unless response_for_current_question.is_a? Hash
```
Comment (kevindew) [inline app/presenters/date_question_presenter.rb:17]: Would we expect this to ever be valid and not be a hash? I wonder if this should just return nil if it's not a hash.

**Comment by leenagupte** [comment]
@kevindew Thanks for the re-review, I've made the suggested change and fixed up.

File: app/presenters/date_question_presenter.rb [hunk 2]
```diff
@@ -15,15 +13,24 @@ def response_label(value)
   end
 
   def selected_day
-    prefill_value_for(self, :day)
+    return if response_for_current_question.blank?
```
Comment (kevindew) [inline app/presenters/date_question_presenter.rb:16]: Minor: you don't actually need these guard conditions now you've got the second one. Sorry I didn't twig on that when I first made the hash suggestion.

**Review by kevindew — APPROVED** [review]
Just a minor thing that I don't think will need another review.

File: app/presenters/date_question_presenter.rb (hunk 2 above)
Comment (leenagupte) [reply app/presenters/date_question_presenter.rb]: Thanks, I've fixed and fixed-up.

**Review by kevindew — APPROVED** [review]

**Pull request merged**